clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp

```diff
@@ -739,6 +795,12 @@ ModuleDepCollectorPP::handleTopLevelModule(const Module *M) {
   MDC.ScanInstance.getASTReader()->visitInputFileInfos(
       *MF, /*IncludeSystem=*/true,
       [&](const serialization::InputFileInfo &IFI, bool IsSystem) {
+        if (MD.IsShareable) {
+          auto FullFilePath = ASTReader::ResolveImportedPath(
+              PathBuf, IFI.UnresolvedImportedFilename, MF->BaseDirectory);
+          MD.IsShareable = isPathInSharedDir(SharedDirs, *FullFilePath);
+          PathBuf.resize_for_overwrite(256);
```

Comment:
Why are you resizing the string here? `ResolveImportedPath` will `clear()` it before using it anyway.

Reply:
I thought I was running into the assertion failure in `ResolveImportedPath`, but I can't reproduce it now.